src/neo-vm/ExecutionEngine.cs

```diff
@@ -892,30 +892,34 @@ private bool ExecuteInstruction()
                             StackItem value = context.EvaluationStack.Pop();
                             if (value is Struct s) value = s.Clone();
                             StackItem key = context.EvaluationStack.Pop();
-                            if (key is ICollection) return false;
                             switch (context.EvaluationStack.Pop())
                             {
                                 case VMArray array:
                                     {
                                         int index = (int)key.GetBigInteger();
                                         if (index < 0 || index >= array.Count) return false;
+
                                         array[index] = value;
+                                        if (!CheckStackSize(true, -3))
```

Comment:
I am not sure, because you can remove a previous item, a map with a lot of items for example.... We should have a faster mechanism for count items, the current solution don't convince me

Reply:
Why `CheckStackSize(true, -3)`?